src/typings/tyranid-extensions.d.ts

```diff
@@ -0,0 +1,147 @@
+         * @param masterKey string key to use for encryption, must be preserved for decryption
+         * 
+         */
+        export function encryptCollection(targetCollection: Tyr.CollectionInstance, encryptionKey: string);
+
+        /**
+         * TODO: Still need to figure out how to tie metadata collection to target
+         *              For now just explicitly pass in collections
+         * TODO: When restoring, what do we want to do with metadata? 
+         * 
+         * Restores data that was masked to its original state
+         * 
+         * @param targetCollection Collection to have data inserted into
+         * @param sourceCollection Collection that data will be pulled from
+         * @param query optionally specify subset of data to restore, if no query specified all records will be replaced
+         * @param decryptionKey Decryption key for the supplied data if it is encrypted.
+         */
+        export function restoreObfuscatedData(targetCollection: Tyr.CollectionInstance, sourceCollection: Tyr.CollectionInstance, query?: Tyr.MongoQuery, decryptionKey?: string)
+        export interface ObfuscateBatchConfig {
+            metadataCollectionName: string
+
+        }
+        export interface ObfuscateBatchResult {
+            /**
+             * Unique identifier for the obfuscate batch.
+             * Set on each metadata record instance to later identify
+             * affected records.
+             */
+            batchTag: ObjectId, //Might use UUID?
+
+            /**
+             * Number of records affected
+             */
+            count: number,
+
+        }
+        export interface ObfuscateBatchOpts {
+            /**
+             * MongoDB query for selecting which records to obfuscate
+             */
+            query: FilterQuery,
+
+            /**
+             * Target Tyranid collection
+             */
+            collection: ObfuscateableCollectionInstance,
+
+            /**
+             * String appended to the given collection name to designate the 
+             * metadata collection
+             */
+            metadataSuffix: string;
+
+            /**
+             * Optionally specify a collecton that contains records with the associated masked records _id field
+             * Fields on the record not specified here will not be altered.
+             * 
+             * [{
+             *      _id: {id of record to alter},
+             *      field: value
+             * },{
+             *      _id: 2,
+             *      name: 'John Doe'
+             * }]
+             * 
+             */
+            replacementValCollection?: Collection,
+
+            /**
+             * Optionally define replacement values for each field of every record
+             * Obfuscated.
+             * {
+             *  field: value
+             * }
+             */
+            replacementValues?: object,
+            
+        }
```

Comment:
Non-blocking:
You could enforce either `replacementValCollection` or `replacementValues` (mutually exclusive) with a base type and type unions:
```typescript
interface OptsCommon {
  query: FilterQuery;
  ...
}
interface OptsWithCol extends OptsCommon {
  replacementValCollection?: Collection;
}
interface OptsWithVals extends OptsCommon {
  replacementValues?: object,
}
export type ObfuscateBatchOpts  = OptsWithCol | OptsWithVals;
```